observer benchmarking

z-a-f opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#29508 observer benchmarking**

Differential Revision: [D18415171](https://our.internmc.facebook.com/intern/diff/D18415171)

Commit:
observer benchmarking

mingzhe09088 commented on benchmarks/operator_benchmark/pt/observer_test.py:
```diff
@@ -0,0 +1,61 @@
+    attrs=[
+        [3, 512, 512, torch.quint8, torch.per_tensor_affine],
+    ],
+
```
remove this line?

mingzhe09088 commented:
Can you rename the test name to qobserver_test.py so it is consistent with other quantized tests?

Commit:
Update on "observer benchmarking"


Differential Revision: [D18415171](https://our.internmc.facebook.com/intern/diff/D18415171)

Commit:
Update on "observer benchmarking"


Differential Revision: [D18415171](https://our.internmc.facebook.com/intern/diff/D18415171)

raghuramank100 commented on benchmarks/operator_benchmark/pt/qobserver_test.py:
```diff
@@ -0,0 +1,60 @@
+    M=[256, 1024],
+    N=[256, 1024],
+    dtype=[torch.quint8],  # dtype doesn't change the timing
+    qscheme=[torch.per_tensor_affine, torch.per_tensor_symmetric,
```
For observers, cross products wont work.  per-channel qconfig only works with per-channel observer for example. I would write this by having one set of configs for per-tensor-quant (HistogramObserver/MinMax/MovingAverage) and another set of configs for per-channel quant (PerChannelMinMax/PerChannelMovingAverage)
We can fix the dtype to be torch.quint8 for all these tests.

raghuramank100 reviewed (changes requested):
We will need to restrict configs so that the tests run, thanks

Commit:
Update on "observer benchmarking"


Differential Revision: [D18415171](https://our.internmc.facebook.com/intern/diff/D18415171)

Commit:
Update on "observer benchmarking"


Differential Revision: [D18415171](https://our.internmc.facebook.com/intern/diff/D18415171)

Commit:
Update on "observer benchmarking"


Differential Revision: [D18415171](https://our.internmc.facebook.com/intern/diff/D18415171)
